piet-cairo/src/lib.rs

```diff
@@ -594,9 +594,9 @@ impl TextLayout for CairoTextLayout {
             // since it's not a hit, check if closer to start or finish
             // and move the appropriate search boundary
             if point.x < grapheme_bounds.leading {
-                right = grapheme_bounds.curr_idx as usize; // should this be -1?
+                right = middle;
             } else if point.x > grapheme_bounds.trailing {
-                left = grapheme_bounds.curr_idx as usize; // should this be +1?
+                left = middle;
```

Comment:
I haven't looked super closely, but just looking at this if/else it isn't clear to me that there can't be cases where we don't match either condition; that is, I can't tell if this is exhaustive, or not.

If it isn't exhaustive, we still risk an infinite loop.

If you're confident it's exhaustive, you might consider adding a final `else { unreachable!("hit_test_point conditional is exhaustive"); }` or similar, or you could consider logging an error and returning `HitTestPoint::default()`.